### ISSUE TITLE
how to create a custom cell in F#

### ISSUE THREAD

**Issue by chinwobble** [SpiegelSoft/XamarinForms.Reactive.FSharp#2]
Hi I am trying to create a custom viewcell to use in a listview.

I created a custom ViewCell type and bound that to the listview but the cells do not seem to be rendering with Xamarin Forms 2.,5.

Any help would be appreciated, thanks.

The listview created in the view looks like this:

```
theme.GenerateListView
          ( this, 
            <@ fun (x:HomeView) -> x.BooksListView @>,
            ListViewCachingStrategy.RecycleElement,
            fun (a:ListView) -> 
              a.IsPullToRefreshEnabled <- true
          )
          |> withEventCommandBinding
            ( this, 
              <@ fun (vm: HomeViewModel) -> vm.RefreshItems @>, 
              <@ fun (v: HomeView) -> v.BooksListView @>,
              "Refreshing")
          |> withEventCommandBinding
            ( this, 
              <@ fun (vm: HomeViewModel) -> vm.SelectItem @>, 
              <@ fun (v: HomeView) -> v.BooksListView @>,
              "ItemSelected")
          |> withTwoWayBinding
            ( this,
              <@ fun (vm: HomeViewModel) -> vm.SelectedBook @>,
              <@ fun (v: HomeView) -> v.BooksListView.SelectedItem @>,
              id,
              id
            )
          |> withTwoWayBinding
            ( this,
              <@ fun (vm: HomeViewModel) -> vm.Refreshing @>,
              <@ fun (v: HomeView) -> v.BooksListView.IsRefreshing @>,
              id,
              id
            )
          |> withItemsSource(this.ViewModel.Items)
          |> withViewCellTemplate(fun () -> 
            let mutable textCell = BookChapterCell()
            theme.GenerateCustomCell(fun i -> textCell <- i)
            |> withOneWayElementBinding(
                textCell, 
                <@ fun (vm: Book) -> vm.Name @>, 
                BookChapterCell.NameProperty, 
                id ) :> ViewCell
          )

```

My Cell looks like this:

```
type BookChapterCell() as this =  
  inherit ViewCell()
  
  let mutable categoryLabel = Unchecked.defaultof<Label> 
  let mutable nameLabel = Unchecked.defaultof<Label>

  do
    let theme = DefaultTheme
    this.View <- theme.GenerateGrid(["Auto"], ["*"; "*"])
    |> withRow(
      [|
        theme.GenerateTitle (fun v -> categoryLabel <- v)
        theme.GenerateLabel (fun v -> nameLabel <- v)
      |]
    )|> createFromRows :> View
    
  static member CategoryProperty =
    BindableProperty.Create("Category", typeof<string>, typeof<BookChapterCell>, "")
  static member NameProperty =
    BindableProperty.Create("Name", typeof<string>, typeof<BookChapterCell>, "")

  member this.Category
    with get () =
      this.GetValue(BookChapterCell.CategoryProperty) :?> string
    and set (value:string) =
      this.SetValue(BookChapterCell.CategoryProperty, value)

  member this.Name
    with get () =
      base.GetValue(BookChapterCell.NameProperty) :?> string
    and set (value:string) =
      base.SetValue(BookChapterCell.NameProperty, value)
```

**Comment by chinwobble** [comment]
I was able to solve this issue. I was missing a set bidning call.

You can close this issue.

**Comment by SpiegelSoft** [comment]
Thanks for that. Sorry I didn't get round to helping you.

I'm doing some ongoing work now on ListView commands and context menus. I hope to add some documentation soon.

**Comment by chinwobble** [comment]
thanks for that.

I'll try to make some contributions to the samples too. 

Can you tell me what the state of each of the sample projects are?
- I've tried to run the Places example but it seems to required an involved setup process or is not functioning.
- How about the Mars sample? Is that something I can add enhancements to?

**Comment by SpiegelSoft** [comment]
I removed the samples from the solution a few months ago. This was purely because of time pressure (I'm using XRF in one of my deliverables and I needed to just get everything working). I'll try and resurrect them this weekend.

The aim of the Mars sample is to get an app into the store. I would definitely like to resurrect it. Along with the Hello World sample, that is going to be my highest priority.
